Improve navigation section pages (#6169)
godotengine/godot-docs: tutorials/navigation/navigation_introduction_2d.rst

.. _doc_navigation_overview_2d:

Introduction
===================

> Godot provides multiple objects, classes and servers to facilitate grid-based- or mesh-based navigation

Reviewer: ```suggestion
Godot provides multiple objects, classes and servers to facilitate grid-based or mesh-based navigation
```

and pathfinding for 2D and 3D games. The following section provides a quick overview over all available

> navigation related objects in Godot for 2D scenes and their primary use.

Reviewer: `object`s is used multiple times and strikes me as a bit odd compared to other Godot docs. Shouldn't we call them nodes and classes? Or are these different?

2D Navigation Overview
----------------------

Godot provides the following objects and classes for 2D navigation:

- :ref:`Astar2D<class_Astar2D>`
    ``Astar2D`` objects provide an option to find the shortest path in a graph of weighted **points**.

    The AStar2D class is best suited for cellbased 2D gameplay that does not require actors to reach any possible position within an area but only predefined, distinct positions.

- :ref:`NavigationServer2D<class_NavigationServer2D>`

>     ``NavigationServer2D`` provides a powerful server API to find the shortest path between two positions on a navigationmesh defined area.

Reviewer: ```suggestion
    ``NavigationServer2D`` provides a powerful server API to find the shortest path between two positions on a area defined by a navigation mesh.
```
`navigationmesh` is used multiple times in the doc, it should IMO either be `navigation mesh`, `navmesh` or link to a class and then be capitalized as `NavigationMesh` if it is one.

            The map will synchronise regions and agents each physics frame.
        - NavRegion RID
            Reference to a specific navigation region that can hold navigationmesh data.
            The region can be enabled / disabled or the use restricted with a navigationlayer bitmask.
        - NavAgent RID
            Reference to a specific avoidance agent with a radius value use solely in avoidance.

The following SceneTree Nodes are available as helpers to work with the NavigationServer2D API.

- :ref:`NavigationRegion2D<class_NavigationRegion2D>` Node
    A Node that holds a NavigationPolygon resource that defines a navigationmesh for the NavigationServer2D.
    The region can be enabled / disabled.
    The use in pathfinding can be further restricted through the navigationlayers bitmask.
    Regions can join their navigationmeshes by proximity for a combined navigationmesh.

-  :ref:`NavigationAgent2D<class_NavigationAgent2D>` Node
    An optional helper Node to facilitate common NavigationServer2D API calls for pathfinding and avoidance
    for a Node2D inheriting parent Node.

-  :ref:`NavigationObstacle2D<class_NavigationObstacle2D>` Node

>     A Node that acts as an agent with avoidance radius, to work it just needs to be added under a Node2D

Reviewer: ```suggestion
    A Node that acts as an agent with avoidance radius, to work it needs to be added under a Node2D
```
